### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by amuraru**
| Q               | A      |
| --------------- | ------ |
| Bug fix?        | no     |
| New feature?    | no     |
| API breaks?     | no     |
| Deprecations?   | no     |
| License         | Apache 2.0 |

### What's in this PR?

Upgrade dependencies

### Why?
Keeping them up to date.

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

**Review by pregnor — CHANGES_REQUESTED** [review]

File: api/go.mod [hunk 1]
```diff
@@ -3,14 +3,14 @@ module github.com/banzaicloud/koperator/api
 go 1.18
 
 require (
-	emperror.dev/errors v0.8.0
+	emperror.dev/errors v0.8.1
 	github.com/banzaicloud/istio-client-go v0.0.17
-	github.com/imdario/mergo v0.3.12
-	github.com/jetstack/cert-manager v1.6.1
+	github.com/cert-manager/cert-manager v1.8.2
+	github.com/imdario/mergo v0.3.13
```
[the remark this answers is not in the document]
Comment (amuraru) [reply api/go.mod:9]: My understanding is that the migration to v1 certmanager APIs was already done in https://github.com/banzaicloud/koperator/commit/9ed415396149bcb12f87166e070361ab57f49f7f

**Review by stoader — APPROVED** [review]
I think references to cert manager needs to be updated in kafka docs to point to the new version: https://github.com/banzaicloud/koperator-docs

File: api/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (stoader) [reply api/go.mod:9]: yes the migration to `v1` was already done

[the remark this answers is not in the document]
Comment (pregnor) [reply api/go.mod:9]: I found a couple references to the removed cert-manager `apiVersion`s: https://github.com/banzaicloud/koperator/search?q=v1alpha2

[the remark this answers is not in the document]
Comment (stoader) [reply api/go.mod:9]: those are used when koperator is deployed using `make deploy` which relies on kustomize. Let's fix those in a separate PR

**Review by pregnor — APPROVED** [review]

**Pull request merged**